### PR TITLE
Convert app/user_test.go t.Fatal calls into assert/require calls

### PR DESCRIPTION
**Pull request by niklabh**
#### Summary

Convert app/user_test.go t.Fatal calls into assert/require calls
#### Ticket Link

Fixes https://github.com/mattermost/mattermost-server/issues/12208

### CODE REVIEW

**Comment by hanzei** [comment]
@levb Gentle reminder to review this PR

File: app/user_test.go [hunk 1]
```diff
@@ -79,41 +79,29 @@ func TestCreateOAuthUser(t *testing.T) {
...
-		t.Fatal(err)
-	}
+	require.Nil(t, err2)
 
 	colorful := color.RGBA{116, 49, 196, 255}
 
-	if img.At(1, 1) != colorful {
-		t.Fatal("Failed to create correct color")
-	}
+	require.Equal(t, img.At(1, 1), colorful, "Failed to create correct color")
```
[the remark this answers is not in the document]
Comment (niklabh) [reply app/user_test.go]: done, for other require.Equal in the file also.

**Comment by levb** [comment]
I kicked the timed-out build, waiting for it to run

**Pull request merged**